torchmd/run.py

```diff
@@ -194,8 +194,15 @@ def setup(args):
             embeddings = torch.tensor(args.external["embeddings"]).repeat(
                 args.replicas, 1
             )
-        output_transform = args.external["output_transform"] if "output_transform" in args.external else None
-        external = externalmodule.External(args.external["file"], embeddings, device, output_transform)
+        # remove from args_external the items that have been already passed to the external module
+        args_external = {
+            key: value
+            for key, value in args_external.items()
+            if key not in ["module", "file", "embedding"]
+        }
+        external = externalmodule.External(
+            args.external["file"], embeddings, device, **args.external
```

Comment:
Change to `..., device=device, **args.external)`. I can see some module having device as the last argument, for instance.